tests/integration/test_charm.py

```diff
@@ -30,6 +30,42 @@ async def test_lifecycle_before_relations(app: ops.model.Application, ops_test:
     unit = app.units[0]  # type: ignore
 
     # Mypy has difficulty with ActiveStatus
-    assert unit.workload_status == ops.model.BlockedStatus.name  # type: ignore
-    # Assert part of the message
-    assert ops.BlockedStatus("Database relation not found") == app.status
+    assert unit.workload_status == ops.model.WaitingStatus.name  # type: ignore
+
+
+@pytest.mark.asyncio
+@pytest.mark.abort_on_fail
+async def test_lifecycle_after_relations(app: ops.model.Application, ops_test: OpsTest):
+    """
+    arrange: build and deploy the charm.
+    act: relate to a db and a matrix homeserver.
+    assert: that the charm ends up in active state.
+    """
+    # Set config so the charm can start
+    config = {"bridge_admins": "admin:example.com", "bot_nickname": "bot"}
+    await tests.integration.helpers.set_config(ops_test, app.name, config)
+    # await ops_test.model.wait_for_idle(apps=[app.name], status="waiting", timeout=60 * 60)
+    # Application actually does have units
+    unit = app.units[0]  # type: ignore
+
+    # Deploy postgresql charm
+    assert ops_test.model
```

Comment:
I think this is not necessary. Having a model is something done by the pytest_operator, and if it does not exists we should not be here.

Reply:
Mypy complains though, caught that  with Greg.

Reply:
Maybe using model as a fixture then instead of ops_test?

Reply:
Done.